### DISCUSSION TITLE
Slow keybinding response for double escape

### DISCUSSION THREAD

**Discussion by MegaBluejay** [xonsh/xonsh#4327] [Q&A]
I'm trying to bind an action to a double press of the escape key in emacs mode, following [the tutorial](https://xon.sh/tutorial_ptk.html). While the binding works, there's a noticeable delay, which isn't present when binding the same action to other combinations, including double presses of other keys. The behavior doesn't change with the eager flag, which makes sense, since I have no other bindings starting with 'escape escape'. Clearly escape is being treated differently, perhaps due to it's double meaning in bindings as an alt modifier.

The two questions I have are
 - What exactly is the cause of this delay?
 - What can be done to bypass it?

Thanks in advance.
[An example config to reproduce](https://gist.github.com/MegaBluejay/2062467e174ece20d1a3e1fe622b67c0)

**Comment by gforsyth** [comment]
Hey @MegaBluejay, sorry for the delay, just noticed this. 
I'm not sure what the issue is with the double press, but it's probably worth asking upstream in prompt_toolkit, since we aren't doing anything additional to the keybindings to handle them.
